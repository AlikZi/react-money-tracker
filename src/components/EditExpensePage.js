import React from "react";
import { connect } from "react-redux";
import ExpenseForm from "./ExpenseForm";
import { startEditExpense, startRemoveExpense } from "../actions/expenses";

export class EditExpensePage extends React.Component {
  onSubmit = expense => {
    this.props.editExpense(this.props.expense.id, expense);
    this.props.history.push("/");
  };
  onClick = () => {
    this.props.removeExpense({ id: this.props.expense.id });
    this.props.history.push("/");
  }
  render() {
    return (
      <div>
      <div className="page-header">
          <div className="container">
            <h1 className="page-header__title">Edit Expense</h1>
          </div>
        </div>
        <ExpenseForm
          expense={this.props.expense}
          onSubmit={this.onSubmit}
        />
        <div className="container">
        <button className="btn btn--secondary btn--remove"
          onClick={this.onClick}
        >
          Remove Expense
        </button>
        </div>
      </div>
    );
  }
}


const mapStateToProps = (state, props) => {
  return {
    expense: state.expenses.find(
      expense => expense.id === props.match.params.id
    )
  };
};

const mapDispatchToProps = dispatch => {
  return {
    editExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
    removeExpense: ({ id }) => dispatch(startRemoveExpense({ id }))
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(EditExpensePage);
