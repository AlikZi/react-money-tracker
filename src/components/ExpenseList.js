import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses';

export const ExpenseList = (props) => (
  <div className="container list">
    <div className="list-header">
      <div className="list-header__expenses">
        Expenses
      </div>
      <div className="list-header__expense">
        Expense
      </div>
      <div className="list-header__amount">
        Amount
      </div>
    </div>
    {props.expenses.length === 0 && <p className="list-item list-item--message">No expenses in this period.</p>}
    {props.expenses.map((expense) => {
      return <ExpenseListItem key={expense.id} {...expense} />;
    })}
  </div>
);

const mapStateToProps = (state) => {
  return {
    expenses: selectExpenses(state.expenses, state.filters)
  };
};

export default connect(mapStateToProps)(ExpenseList);
