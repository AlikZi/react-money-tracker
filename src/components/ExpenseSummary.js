import React from 'react';
import getExpensesTotal from './expenses-total';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import selectExpenses from '../selectors/expenses';
import numeral from 'numeral';

export const ExpensesSummary = (props) => {
    const expensesCount = props.expenses.length;
    const expensesWord = props.expenses.length === 1 ? 'expense': 'expenses';
    const expensesTotal = numeral(getExpensesTotal(props.expenses)/100).format('$0,0.00');
    
    return (<div className="summary">
        <div className="container">
        <div className="summary__content">
        <h2 className="summary__content-text">Viewing <span>{expensesCount}</span> {expensesWord} totaling <span>{expensesTotal}</span></h2>
        <Link to='/create' className="btn summary__content-btn">Add Expense</Link>
        </div>
        </div>
    </div>)
}

const mapStateToProps = (state) => {
    return {
      expenses: selectExpenses(state.expenses, state.filters)
    };
  };

export default connect(mapStateToProps)(ExpensesSummary);