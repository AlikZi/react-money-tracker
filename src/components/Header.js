import React from 'react';
import { Link } from 'react-router-dom';
import { startLogout } from '../actions/auth';
import { connect } from 'react-redux';

export const Header = (props) => (
  <header className="header">
    <div className="container">
    <div className="header__content">
    <Link className="header__content-title" to="/dashboard">
      <h1>Money Tracker</h1>
    </Link>

    <button className="link-btn" onClick={props.startLogout}>Logout</button>
    </div>
    </div>  
  </header>
);

const mapDispatchToProps = (dispatch) => ({
  startLogout: () => dispatch(startLogout())
})

export default connect(undefined, mapDispatchToProps)(Header);
