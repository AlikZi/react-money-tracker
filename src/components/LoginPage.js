import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

export const LoginPage = (props) => (
    <div className='box-layout'>
        <div className="box-layout__box">
            <h1 className="box-layout__box-title">Money Tracker</h1>
            <p className="box-layout__box-text">It's time to get your expenses under control.</p>
            <button className="btn btn--login" onClick={props.startLogin}>Login with Google</button>
        </div>
    </div> 
);

const mapDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin())
})

export default connect(undefined, mapDispatchToProps)(LoginPage);