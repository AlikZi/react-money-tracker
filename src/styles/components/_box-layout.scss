.box-layout {
    width: 100vw;
    height: 100vh;
    background: url('/images/bg.jpg');
    background-size: cover;
    display: flex;
    justify-content: center;
    align-items: center;


    &__box {
        background-color: rgba(white, .85);
        padding: 4rem 3rem;
        text-align: center;
        border-radius: 5px;
        width: 25rem;

        &-title{
            margin: 0;
            margin-bottom: $s-size;
            line-height: 1;
        }

        &-text{
            margin-bottom: $l-size;
        }

    }
}

