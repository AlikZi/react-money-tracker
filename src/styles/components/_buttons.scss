.btn {
  padding: 1rem 2rem;
  border-radius: 4px;
  border: none;
  outline: none;
  font-size: 2rem;
  background-color: $color-primary;
  color: #fff;
  text-decoration: none;
  font-weight: 300;
  line-height: 1.2;
  transition: all .2s ease-out;

  &--remove {
    margin: -2rem 0 5rem 0;
  }
}

.btn:hover {
  background-color: darken($color-primary, 5%);
}

.btn:active {
  background-color: darken($color-primary, 10%);
}

.btn--secondary {
  background-color: darken($grey-light, 40%);

  &:hover {
    background-color: darken($grey-light, 45%);
  }
  
  &:active {
    background-color: darken($grey-light, 50%);
  }

}

.link-btn {
  border: none;
  outline: none;
  color: white;
  background-color: transparent;
  font-size: 1.6rem;
  font-weight: 200;
  padding: $s-size 0;
}

