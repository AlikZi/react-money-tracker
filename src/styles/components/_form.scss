.page-header{
    background-color: $grey-light;

    &__title{
        margin: 0;
        padding: $m-size 0;
        font-weight: 100;
        font-size: 3rem;
    }
}

.form{
    margin: $l-size 0;
    display: flex;
    width: 50%;
    flex-direction: column;

    @media only screen and (max-width: $desktop-breakpoint) {
        width: 100%;
    }
    

    & > * {
        margin-bottom: $s-size;
    }

    &__error {
        margin: 0 0 $m-size 0;
        font-style: italic;
        color:#cc3300;
    }

    &__btn{
        margin: 2rem 0 0 0;
    }
}