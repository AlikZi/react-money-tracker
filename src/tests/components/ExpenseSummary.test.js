import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpenseSummary';
import expenses from '../fixtures/expenses';

test('should correctly render ExpensesSummary wiht 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expenses={[expenses[0]]}/>);
    expect(wrapper).toMatchSnapshot();
})

test('should correctly render ExpensesSummary wiht multiple expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenses={expenses}/>);
    expect(wrapper).toMatchSnapshot();
})