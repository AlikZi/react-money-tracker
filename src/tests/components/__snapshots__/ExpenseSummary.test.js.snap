// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpensesSummary wiht 1 expense 1`] = `
<div
  className="summary"
>
  <div
    className="container"
  >
    <div
      className="summary__content"
    >
      <h2
        className="summary__content-text"
      >
        Viewing 
        <span>
          1
        </span>
         
        expense
         totaling 
        <span>
          $1.95
        </span>
      </h2>
      <Link
        className="btn summary__content-btn"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`should correctly render ExpensesSummary wiht multiple expenses 1`] = `
<div
  className="summary"
>
  <div
    className="container"
  >
    <div
      className="summary__content"
    >
      <h2
        className="summary__content-text"
      >
        Viewing 
        <span>
          3
        </span>
         
        expenses
         totaling 
        <span>
          $1,142.45
        </span>
      </h2>
      <Link
        className="btn summary__content-btn"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
